aws/rust-runtime/aws-sig-auth/src/signer.rs

```diff
@@ -159,19 +167,24 @@ impl SigV4Signer {
 
         let (signing_instructions, signature) = {
             // A body that is already in memory can be signed directly. A  body that is not in memory
-            // (any sort of streaming body) will be signed via UNSIGNED-PAYLOAD.
-            let signable_body = request_config
-                .payload_override
-                // the payload_override is a cheap clone because it contains either a
-                // reference or a short checksum (we're not cloning the entire body)
-                .cloned()
-                .unwrap_or_else(|| {
-                    request
-                        .body()
-                        .bytes()
-                        .map(SignableBody::Bytes)
-                        .unwrap_or(SignableBody::UnsignedPayload)
-                });
+            // (any sort of streaming body or presigned request) will be signed via UNSIGNED-PAYLOAD.
+            let signable_body =
```

Comment:
can you log a warning, either here or in `aws-sigv4` if the credentials expiry comes before the signature expiry?

Reply:
Good call! Added.